af3_mmseqs2/processoutput/utils.py

```diff
@@ -0,0 +1,91 @@
+import json
+import logging
+from abc import ABC
+from enum import Enum
+from pathlib import Path
+from typing import Union
+
+import gemmi
+import numpy as np
+
+logger = logging.getLogger("logger")
+
+
+class FileTypes(Enum):
+    NPZ = "npz"
+    CIF = "cif"
+    JSON = "json"
+
+
+class ModelCount:
+    ALL = "all"
+    RESIDUES = "residues"
+
+
+class FileBase(ABC):
+
+    def __init__(self, pathway: Union[str, Path]):
+        self.pathway = Path(pathway)
+        self.suffix = self.pathway.suffix[1:]
+
+    def __str__(self):
+        return str(self.pathway)
+
+    def __repr__(self):
+        return f"{self.__class__.__name__}({self.pathway})"
+
+
+class NpzFile(FileBase):
+    def __init__(self, npz_file: Union[str, Path]):
+        super().__init__(npz_file)
+        self.npz_file = Path(npz_file)
+        self.data = self.load_npz_file()
+
+    def load_npz_file(self):
+        return dict(np.load(self.npz_file))
+
+
+class CifFile(FileBase):
+    def __init__(self, cif_file: Union[str, Path]):
+        super().__init__(cif_file)
+        self.cif_file = Path(cif_file)
+        self.model = self.load_cif_file()
+
+    def load_cif_file(self):
+        # load the cif file
+        return gemmi.read_structure(str(self.cif_file))
```

Comment:
Maybe better to do this with Biopython?

I used gemmi originally to provide templates to AF3 and it didn't like something about how they were formated.

Reply:
It can still do the same things but if using Bio we could use this to make templates too

Reply:
If I can help it I want to avoid biopython but if you wanted to use biopython for the templates I'm sure there is a gemmi model class to biopython converter.

Reply:
I've updated to use Biopython now.